plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -4862,10 +4862,15 @@ public synchronized Network setupvSwitchNetwork(final Connection conn) {
         return null;
     }
 
-    public void shutdownVM(final Connection conn, final VM vm, final String vmName) throws XmlRpcException {
+    public void shutdownVM(final Connection conn, final VM vm, final String vmName, final boolean forcedStop) throws XmlRpcException {
         Task task = null;
         try {
-            task = vm.cleanShutdownAsync(conn);
+            if (forcedStop) {
+                task = vm.hardShutdownAsync(conn);
```

Comment:
@rhtyd , If there is an exception in doing a hard-shutdown on the VM, the logger in 'catch' would still log it as "Unable to cleanShutdown VM". May be should consider changing the log statement accordingly.